src/NLog/Targets/TargetWithContext.cs

```diff
@@ -286,33 +286,48 @@ protected bool ShouldIncludeProperties(LogEventInfo logEvent)
         {
             if (EstimateAllPropertiesCount(logEvent, out var logEventProperties, out var scopeContextProperties, out var gdcNames, out int propertiesCount) > 0)
             {
-                List<KeyValuePair<string, object>> combinedProperties = new List<KeyValuePair<string, object>>(propertiesCount);
                 bool checkExcludeProperties = ExcludeProperties.Count > 0;
-
-                Action<string, object> tryAddProperty = (string propertyName, object propertyValue) =>
+                Func<string, object, bool, KeyValuePair<string, object> ?> buildPair = (string propertyName, object propertyValue, bool serialize) =>
```

Comment:
Possible this can be moved into a private-method, instead of local-method ?

Reply:
moved to `GetAllPropertiesCountContext` class  (explanation below)